pkg/apis/batch/v1/defaults.go

```diff
@@ -39,4 +39,12 @@ func SetDefaults_Job(obj *Job) {
 		obj.Spec.Parallelism = new(int32)
 		*obj.Spec.Parallelism = 1
 	}
+	// When manualSelector is true, set metadata.Labels according to
+	// obj.Spec.Template.Labels
+	labels := obj.Spec.Template.Labels
+	if labels != nil {
+		if len(obj.Labels) == 0 {
+			obj.Labels = labels
+		}
+	}
```

Comment:
I'm not sure we want to copy pod's template labels to the job, if none are found. I need to dig into what's the root cause of those missing labels. But at this point this is not what I'd expect it to be.